Fix and modification of "download_file"

solanav opened the pull request:
There was an error when downloading text files on `f.write(res.data)` in webdav.py:
```
TypeError: a bytes-like object is required, not 'str'
```

I also added a two new parameters to this function:
- download_path (str): to specify where you want to download the file instead of assuming "./"
- overwrite (bool): to choose if you want to overwrite files or you want to raise an exception on duplicate files.

Commit:
Fixed error when downloading text files and added more options to download function

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/34c7ba85-8718-409f-89c2-3bfbf6342d7a) completed: **1 new issues, 1 updated code elements**

Commit:
Added unquote(filename)

Filenames are URL-encoded, so we should preserve the original filename when saving them.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/7ac76581-4a27-439e-9ce8-0f7a2c9a3add) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/7ac76581-4a27-439e-9ce8-0f7a2c9a3add) completed: **1 new issues, 1 updated code elements**

luffah commented:
( If you still have the test case, i ll be grateful if you test with my repo — refactored code and python2 compat — : https://github.com/luffah/nextcloud-API )
